plugins/elasticsearch-bboss/src/main/java/com/navercorp/pinpoint/plugin/elasticsearchbboss/ElasticsearchPlugin.java

```diff
@@ -0,0 +1,202 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.navercorp.pinpoint.plugin.elasticsearchbboss;
+
+import com.navercorp.pinpoint.bootstrap.instrument.InstrumentClass;
+import com.navercorp.pinpoint.bootstrap.instrument.InstrumentException;
+import com.navercorp.pinpoint.bootstrap.instrument.InstrumentMethod;
+import com.navercorp.pinpoint.bootstrap.instrument.Instrumentor;
+import com.navercorp.pinpoint.bootstrap.instrument.transformer.TransformCallback;
+import com.navercorp.pinpoint.bootstrap.instrument.transformer.TransformTemplate;
+import com.navercorp.pinpoint.bootstrap.instrument.transformer.TransformTemplateAware;
+import com.navercorp.pinpoint.bootstrap.interceptor.scope.ExecutionPolicy;
+import com.navercorp.pinpoint.bootstrap.logging.PLogger;
+import com.navercorp.pinpoint.bootstrap.logging.PLoggerFactory;
+import com.navercorp.pinpoint.bootstrap.plugin.ProfilerPlugin;
+import com.navercorp.pinpoint.bootstrap.plugin.ProfilerPluginSetupContext;
+import com.navercorp.pinpoint.common.trace.*;
+
+import java.security.ProtectionDomain;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.navercorp.pinpoint.common.trace.ServiceTypeProperty.RECORD_STATISTICS;
+
+/**
+ * @author yinbp[yin-bp@163.com]
+ */
+public class ElasticsearchPlugin implements ProfilerPlugin, TransformTemplateAware {
+	public static final AnnotationKey ARGS_ANNOTATION_KEY = AnnotationKeyFactory.of(971, "es.args", AnnotationKeyProperty.VIEW_IN_RECORD_SET);
+
+	public static final ServiceType ELASTICSEARCH = ServiceTypeFactory.of(1971, "ElasticsearchBBoss", RECORD_STATISTICS);
+
+	public static final ServiceType ELASTICSEARCH_EVENT = ServiceTypeFactory.of(1972, "ElasticsearchBBossEvent");
+	public static final AnnotationKey ARGS_URL_ANNOTATION_KEY = AnnotationKeyFactory.of(972, "es.url", AnnotationKeyProperty.VIEW_IN_RECORD_SET);
+	public static final AnnotationKey ARGS_DSL_ANNOTATION_KEY = AnnotationKeyFactory.of(973, "es.dsl", AnnotationKeyProperty.VIEW_IN_RECORD_SET);
+	public static final AnnotationKey ARGS_ACTION_ANNOTATION_KEY = AnnotationKeyFactory.of(974, "es.action", AnnotationKeyProperty.VIEW_IN_RECORD_SET);
+	public static final AnnotationKey ARGS_RESPONSEHANDLE_ANNOTATION_KEY = AnnotationKeyFactory.of(975, "es.responseHandle", AnnotationKeyProperty.VIEW_IN_RECORD_SET);
+	public static final AnnotationKey ARGS_VERSION_ANNOTATION_KEY = AnnotationKeyFactory.of(976, "es.version", AnnotationKeyProperty.VIEW_IN_RECORD_SET);
+	public static final ServiceType ELASTICSEARCH_EXECUTOR = ServiceTypeFactory.of(9977, "ElasticsearchBBossExecutor");
+	private static final String ELASTICSEARCH_SCOPE = "ElasticsearchBBoss_SCOPE";
+	private static final String ELASTICSEARCH_EXECUTOR_SCOPE = "ElasticsearchBBossExecutor_SCOPE";
+	private static final List<ElasticsearchInterceptorClassInfo> clazzInterceptors = new ArrayList<ElasticsearchInterceptorClassInfo>();
+
+	static {
+		init();
```

Comment:
I wonder if this static method is necessary.
I think what you are trying to do is, getting the methods that need to be instrumented.
Correct me if I'm wrong.

I think it will be best to check how other plugins listed their _methods_ that need to be instrumented.

and usually, it's best to list them(ex. CssandraPlugin.java) or use for loops(ex. _InstrumentMethod method : target.getDeclaredMethods(methodFilter)_ in ActiveMQClientPlugin.java).

Reply:
Thank for your suggests. Elasticsearchbboss plugin is writed based on [pinpoint-plugin-generate](https://github.com/bbossgroups/pinpoint-plugin-generate) tool,I'll refacted it soon as your suggest.